### ISSUE TITLE
[behavior-change] D202 don't flag if there is a leading comment

### ISSUE THREAD

**Issue by smackesey** [charliermarsh/ruff#1327]
Upgraded to black 22.12.0 (preview mode) and noticed that ruff now disagrees with black over how to format this:

```
def analytics_call(fn):
    """Decorator for all functions below that write to segment.com."""

    # Catch and log errors but don't propagate. We don't want analytics write failure to bubble up
    # to user actions.
    # Ensure write_key is present or log warning.
```

Ruff `D202` wants to get rid of the blank line after docstring, but black requires that it be there (only since the next bit of code is a comment). I don't know how Flake8 does it but I think black's way makes sense here.

**Comment by charliermarsh** [comment]
Oh, interesting, so it wants a blank line when it's followed by a comment? We can do that.

**Comment by charliermarsh** [comment]
Confusingly, if I run `black --preview` on 22.12.0 over this code:

```py
def analytics_call(fn):
    """Decorator for all functions below that write to segment.com."""
    # Catch and log errors but don't propagate. We don't want analytics write failure to bubble up
    # to user actions.
    # Ensure write_key is present or log warning.
```

It doesn't add that newline. Is that what you're seeing?

**Comment by charliermarsh** [comment]
Closing for now, Black seems ok with either.

**Comment by smackesey** [comment]
Hey sorry for not following up here, I'm not sure what exactly was going on but the issue seems resolved now in any case.

**Comment by charliermarsh** [comment]
All good, no prob :)
